mne/io/reference.py

```diff
@@ -321,13 +322,19 @@ def set_eeg_reference(inst, ref_channels='average', copy=True,
             # Creating an average reference may fail. In this case, make
             # sure that the custom_ref_applied flag is left untouched.
             custom_ref_applied = inst.info['custom_ref_applied']
+
+            if len(ch_type) > 1:
+                assert len(ch_type_orig) > 1  # only have 2 if 2 were passed
+                raise ValueError('Multiple channel types are not supported '
+                                 f'when using projection. Got {ch_type_orig}.')
```

Comment:
Dear @agramfort, thank you very much for working on this! In my case, I would actually love to use the average reference projection for multiple channel types. 

My patient dataset has both ECOG strips on the cortex and DBS strips in deep brain structures. I would like to apply an average reference projection for both channel types `ch_type=('ecog', 'dbs')` together. It seems, you would like to explicitly **exclude** this possibility. 

Would it be possible to **include** this possibility?

Reply:
hum yes for me it was dangerous to do this as different channel types easily have different units and it can be error prone but I get your usecase. We would have to see if it breaks somewhere else if we do this... can you give it a try on top of my commits here?